### ISSUE TITLE
Improvement of task management 

### ISSUE THREAD

**Issue by makorne** [Nukesor/pueue#397]
#### Is your feature request related to a problem?

I try to create many long-living tasks.
Each of them has its own unique Id like 11111 but I cant find how to use these ids for pueue task_id creating.

I see Labels in v3.0 but it cant be used for managing tasks but only for status:

`v3.0.0 also adds the long-requested feature to add a query/filter logic for the status command`

#### Describe the solution you'd like

Add the possibility to create task with a custom unique id. 
May be for a certain Group only. 

Or add a possibility of managing (kill/pause/restart) tasks by Label.

**Comment by Nukesor** [comment]
What's your exact use case for pueue?
It sounds like you're trying to use it for scripting? In that case, you can use he --print-task-id flag on the `add` command, to get the id of a specific task. You can lateron use that Id to work with it.

**Comment by makorne** [comment]
Ok.
Thank you for your great help and work!

**Comment by mantljosh** [comment]
@Nukesor any chance of reconsidering this? My use case is for local development on a monorepo project where we can't use Docker as it is horrendously slow on OS X - so instead we run our services un-containerized during local development.

I want to be able to reference a long-running task (i.e. the service process) by name when I'm starting/logging/restarting etc. as I only want one instance of each, and task IDs are different every time depending on the order I end up starting services, so there's no way to have a stable command that always targets the same logical task.

Labels seem to be the closest thing right now, but I think really what I'd ideally want is the ability to specify the task ID so I can ensure a stable ID to reference the same logical tasks.

So far `pueue` is the closest tool to `pm2` (which has been nothing but trouble for me) I've found, but if there are other tools you think are better suited for this I appreciate any suggestions.

**Comment by Nukesor** [comment]
Yep. Your use-case is something that sadly isn't something Pueue was designed for.

Tasks that're added to Pueue are supposed to end on their own at some point. It's basically a task queue  with a very simple (by design) scheduler on top.

You're using it as a process manager, which isn't something it was built for.
`pm2` is actually one of the solutions for your problems.

I've worked of another tool like this, but I don't remember the name of it right now :sweat: 
I'll write as soon as I stumble upon it or if my brain starts working again ;D

**Comment by mantljosh** [comment]
Thanks for the reply and the suggestions! Took a look at those (and some others linked from those projects) and definitely some similarities, but none quite doing what PM2 aims to do.

> I would actually be up to write a nice alternative to pm2, if I find the time to do so

I'm considering the same, but like you "if I find the time" is the tricky part :)
